scripts/bump_deps.lua

```diff
@@ -0,0 +1,378 @@
+run({ "mkdir", "-p", temp_dir })
+local gh_res_path = temp_dir .. "/gh_res.json"
+
+local function get_dependency(dependency_name)
+	local dependency_table = {
+		["LuaJIT"] = {
+			repo = "LuaJIT/LuaJIT",
+			symbol = "LUAJIT",
+		},
+		["libuv"] = {
+			repo = "libuv/libuv",
+			symbol = "LIBUV",
+		},
+		["Luv"] = {
+			repo = "luvit/luv",
+			symbol = "LUV",
+		},
+		["tree-sitter"] = {
+			repo = "tree-sitter/tree-sitter",
+			symbol = "TREESITTER",
+		},
+	}
+	local dependency = dependency_table[dependency_name]
+	if dependency == nil then
+		p("Not a dependency: " .. dependency_name)
+		die()
+	end
+	dependency.name = dependency_name
+	return dependency
+end
+
+local function dl_gh_ref_info(repo, ref)
+	require_executable("curl")
+
+	rm_file_if_present(gh_res_path)
+	local ref_status_code = run_die({
+		"curl",
+		"-s",
+		"-H",
+		"'Accept: application/vnd.github.v3+json'",
+		"-w",
+		"'%{http_code}'",
+		"https://api.github.com/repos/" .. repo .. "/commits/" .. ref,
+		"-o",
+		gh_res_path,
+	}, "Failed to fetch commit details from GitHub")
+	if ref_status_code ~= "'200'" then
+		p("Not a valid ref: " .. ref)
+		die()
+	end
+end
+
+local function get_json_field(jsonfile_path, field)
+	require_executable("jq")
+	return run({ "jq", "-r", "." .. field, jsonfile_path })
+end
+
+local function get_archive_info(repo, ref)
+	require_executable("curl")
+
+	local archive_name = ref .. ".tar.gz"
+	local archive_path = temp_dir .. "/" .. archive_name
+	local archive_url = "https://github.com/" .. repo .. "/archive/" .. archive_name
+
+	rm_file_if_present(archive_path)
+	run_die({ "curl", "-sL", archive_url, "-o", archive_path }, "Failed to download archive from GitHub")
+
+	local archive_sha = run({ "sha256sum", archive_path }):gmatch("%w+")()
+	return { url = archive_url, sha = archive_sha }
+end
+
+local function write_cmakelists_line(symbol, kind, value)
+	require_executable("sed")
+
+	local cmakelists_path = nvim_src_dir .. "/" .. "third-party/CMakeLists.txt"
+	run_die({
+		"sed",
+		"-i",
+		"-e",
+		"s/set(" .. symbol .. "_" .. kind .. ".*$" .. "/set(" .. symbol .. "_" .. kind .. " " .. value .. ")" .. "/",
+		cmakelists_path,
+	}, "Failed to write " .. cmakelists_path)
+end
+
+local function update_cmakelists(dependency, archive, comment)
+	local changed_file = nvim_src_dir .. "/" .. "third-party/CMakeLists.txt"
+
+	p("Updating " .. dependency.name .. " to " .. archive.url .. "\n")
+	write_cmakelists_line(dependency.symbol, "URL", archive.url:gsub("/", "\\/"))
+	write_cmakelists_line(dependency.symbol, "SHA256", archive.sha)
+	run_die(
+		{ "git", "commit", changed_file, "-m", commit_prefix .. "bump " .. dependency.name .. " to " .. comment },
+		"git failed to commit"
+	)
+end
+
+local function verify_cmakelists_committed()
+	local cmakelists_path = nvim_src_dir .. "/" .. "third-party/CMakeLists.txt"
+	run_die({ "git", "diff", "--quiet", "HEAD", "--", cmakelists_path }, cmakelists_path .. " has uncommitted changes")
```

Comment:
I think this needs to be against `origin/master` instead, and it might be safer to use the plumbing command `diff-index` as well.

Reply:
It's ok to leave it like it is.